core/src/main/java/com/linecorp/armeria/common/metric/MeterIdPrefixFunction.java

```diff
@@ -73,15 +76,21 @@ static MeterIdPrefixFunction ofDefault(String name) {
                 methodName = MoreObjects.firstNonNull(log.method().name(), "__UNKNOWN_METHOD__");
             }
 
+            List<Tag> tags = new ArrayList<>();
```

Comment:
this will increase memory footprint a bit, but without the List, I can not think about any another way without a lot of if/else. Could anybody give me some advice?

Reply:
- I think it's fine. If escape analysis works as expected, it should be stack-allocated. You could specify the initial capacity correctly though. How about 4 (method, hostnamePattern, pathMapping, status)?
- Please make `tags` final.

Reply:
not sure List<Tag> will be stack-allocated until it will be escaped to return method, anyway specify initial cap SGTM, and added

Reply:
`MeterIdPrefix` will make a copy of it so it can stay at stack if the constructor of `MeterIdPrefix` is inlined.